packages/examples/main/src/browser/main.ts

```diff
@@ -11,145 +11,153 @@ import { buildWorkerDefinition } from 'monaco-editor-workers';
 import { getLanguageService, TextDocument } from 'vscode-json-languageservice';
 import { createConverter as createCodeConverter } from 'vscode-languageclient/lib/common/codeConverter.js';
 import { createConverter as createProtocolConverter } from 'vscode-languageclient/lib/common/protocolConverter.js';
-
-import { StandaloneServices } from 'vscode/services';
-import getNotificationServiceOverride from 'vscode/service-override/notifications';
-import getDialogServiceOverride from 'vscode/service-override/dialogs';
-
-StandaloneServices.initialize({
-    ...getNotificationServiceOverride(document.body),
-    ...getDialogServiceOverride()
-});
+import { initServices } from 'monaco-languageclient';
+import { createConfiguredEditor, createModelReference } from 'vscode/monaco';
 
 buildWorkerDefinition('../../../node_modules/monaco-editor-workers/dist/workers/', new URL('', window.location.href).href, false);
 
 const codeConverter = createCodeConverter();
 const protocolConverter = createProtocolConverter(undefined, true, true);
 
 const LANGUAGE_ID = 'json';
-const MODEL_URI = 'inmemory://model.json';
+const MODEL_URI = '/tmp/model.json';
 const MONACO_URI = monaco.Uri.parse(MODEL_URI);
 
-// register the JSON language with Monaco
-monaco.languages.register({
-    id: LANGUAGE_ID,
-    extensions: ['.json', '.jsonc'],
-    aliases: ['JSON', 'json'],
-    mimetypes: ['application/json']
-});
+const createEditor = async () => {
+    // register the JSON language with Monaco
+    monaco.languages.register({
+        id: LANGUAGE_ID,
+        extensions: ['.json', '.jsonc'],
+        aliases: ['JSON', 'json'],
+        mimetypes: ['application/json']
+    });
 
-// create the Monaco editor
-const value = `{
+    // create the Monaco editor
+    const content = `{
     "$schema": "http://json.schemastore.org/coffeelint",
     "line_endings": "unix"
 }`;
-const model = monaco.editor.createModel(value, LANGUAGE_ID, MONACO_URI);
-monaco.editor.create(document.getElementById('container')!, {
-    model,
-    glyphMargin: true,
-    lightbulb: {
-        enabled: true
-    },
-    automaticLayout: true
-});
+    const modelRef = await createModelReference(MONACO_URI, content);
+    modelRef.object.setLanguageId(LANGUAGE_ID);
+    createConfiguredEditor(document.getElementById('container')!, {
+        model: modelRef.object.textEditorModel,
+        glyphMargin: true,
+        lightbulb: {
+            enabled: true
+        },
+        automaticLayout: true
+    });
 
-const vscodeDocument = vscode.workspace.textDocuments[0];
+    const mainVscodeDocument = await vscode.workspace.openTextDocument({
```

Comment:
@CGNonofr this now longer works
```javascript
const vscodeDocument = vscode.workspace.textDocuments[0];
```
But my replacement throws an error:
![image](https://user-images.githubusercontent.com/21079138/235684837-89f240d6-768f-4596-9216-a8ebeb1f59c2.png)
Do you have an idea how to fix this? Will https://github.com/CodinGame/monaco-vscode-api/pull/109 be of help here, too?

Reply:
Can you please add this comment on the other side (on the PR) to not forget about it?

Reply:
Btw i'm not sure to understand why you want to open an untitled text document

Reply:
Like this? https://github.com/CodinGame/monaco-vscode-api/pull/109#issuecomment-1531551700

Reply:
> Btw i'm not sure to understand why you want to open an untitled text document

It seems in the past a default document was available. Is there a way to create a `vscode.TextDocument` from the already existing textEditorModel or is it actually wrong to use the workspace here.

The real question is if it is worth to keep the browser example at all if we have web worker based language servers now, but still I want to understand/fix this problem here?

Reply:
vscode.workspace.textDocuments returns a vscode.TextDocument for each open model, I don't know what what that "default" document that was there before

maybe you're calling it too early, the synchronization between the extHost and the services was made asynchronous, maybe you can wait on `vscode.workspace.onDidOpenTextDocument`

Reply:
> maybe you're calling it too early, the synchronization between the extHost and the services was made asynchronous, maybe you can wait on vscode.workspace.onDidOpenTextDocument

This could explain it. I will give it a try later.

Reply:
Fixed it, see below ⬇️